web-marketplace/src/pages/Marketplace/Storefront/Storefront.tsx

```diff
@@ -256,62 +242,30 @@ export const Storefront = (): JSX.Element => {
-                              setSelectedSellOrder(i);
-                            },
-                          },
-                        ]}
-                        sx={{ width: '100%' }}
-                      />
-                    )}
-                    {!isOwnSellOrder && (
-                      <OutlinedButton
-                        startIcon={<CreditsIcon className="text-brand-400" />}
-                        size="small"
-                        onClick={async () => {
-                          track<Buy1Event>('buy1', {
-                            url: location.pathname,
-                            buttonLocation: 'sellOrderTable',
-                            projectName: normalizedSellOrders[i].project?.name,
-                            projectId: normalizedSellOrders[i].project?.id,
-                            creditClassId:
-                              normalizedSellOrders[i].project?.classId,
-                          });
-                          selectedSellOrderIdRef.current = Number(
-                            normalizedSellOrders?.[i].id,
-                          );
-                          submittedQuantityRef.current = undefined;
-                          refetchSellOrders();
-                          setSelectedAction('buy');
-                          setSelectedSellOrder(i);
-                          if (!activeWalletAddr) {
-                            setConnectWalletModalAtom(
-                              atom => void (atom.open = true),
-                            );
-                          } else if (!isConnected) {
-                            setSwitchWalletModalAtom(
-                              atom => void (atom.open = true),
-                            );
-                          }
-                        }}
-                      >
-                        {_(BUY_SELL_ORDER_ACTION)}
-                      </OutlinedButton>
-                    )}
-                  </>
-                );
-              }}
+              renderActionButtonsFunc={
+                isConnected
+                  ? (i: number) => {
+                      const isOwnSellOrder =
+                        normalizedSellOrders[i]?.seller === accountAddress;
+                      return (
+                        isOwnSellOrder && (
+                          <TableActionButtons
+                            buttons={[
+                              {
+                                label: _(CANCEL_SELL_ORDER_ACTION),
+                                onClick: () => {
+                                  setSelectedAction('cancel');
+                                  setSelectedSellOrder(i);
+                                },
+                              },
+                            ]}
+                            sx={{ width: '100%' }}
+                          />
+                        )
+                      );
+                    }
+                  : undefined
+              }
```

Comment:
we can also get rid of `BuyCreditsModal` and related code: `isBuyModalOpen`, etc.

Reply:
Thanks for pointing that out